code/modules/mob/vision_cone.dm

```diff
@@ -0,0 +1,116 @@
+/////////////VISION CONE///////////////
+//Vision cone code by Matt and Honkertron. This vision cone code allows for mobs and/or items to blocked out from a players field of vision.
+//This code makes use of the "cone of effect" proc created by Lummox, contributed by Jtgibson. More info on that here:
+//http://www.byond.com/forum/?post=195138
+///////////////////////////////////////
+
+//"Made specially for Otuska"
+// - Honker
+
+
+
+//Defines.
+/mob
+	var/obj/screen/fov = null//The screen object because I can't figure out how the hell TG does their screen objects so I'm just using legacy code.
+
+client/
+	var/list/hidden_atoms = list()
+	var/list/hidden_mobs = list()
+
+
+
+atom/proc/InCone(atom/center = usr, dir = NORTH)
+	if(get_dist(center, src) == 0) return 0
+	var/d = get_dir(center, src)
+
+	if(!d || d == dir) return 1
+	if(dir & (dir-1))
+		return (d & ~dir) ? 0 : 1
+	if(!(d & dir)) return 0
+	var/dx = abs(x - center.x)
+	var/dy = abs(y - center.y)
+	if(dx == dy) return 1
+	if(dy > dx)
+		return (dir & (NORTH|SOUTH)) ? 1 : 0
+	return (dir & (EAST|WEST)) ? 1 : 0
+
+mob/dead/InCone(mob/center = usr, dir = NORTH)
+	return
+
+mob/living/InCone(mob/center = usr, dir = NORTH)
+	if(get_dist(center, src) == 0 || src == center) return 0
+	//for(var/obj/item/weapon/grab/G in center)//TG doesn't have the grab item. But if you're porting it and you do then uncomment this.
+	//	if(src == G.affecting) return 0
+	var/d = get_dir(center, src)
+
+	if(!d || d == dir) return 1
+	if(dir & (dir-1))
+		return (d & ~dir) ? 0 : 1
+	if(!(d & dir)) return 0
+	var/dx = abs(x - center.x)
+	var/dy = abs(y - center.y)
+	if(dx == dy) return 1
+	if(dy > dx)
+		return (dir & (NORTH|SOUTH)) ? 1 : 0
+	return (dir & (EAST|WEST)) ? 1 : 0
+
+proc/cone(atom/center = usr, dir = NORTH, list/list = oview(center))
+    for(var/atom/O in list) if(!O.InCone(center, dir)) list -= O
+    return list
```

Comment:
The reason for this is because this is apart of the ["Cone of Affect" code](http://www.byond.com/forum/?post=195138l) that Lummox wrote that I used as the basis for this. Neither me nor Honker touched this part of the code.